### PR TITLE
Crystallized cerulean tweaks

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

Crystallized cerulean only spawns 3 crystals.
Each crystal's grow time is doubled (from 60 to 120).
Each crystal now drops the crystal (yes) after it reaches stage 4.
Stage 4 crystals adds 1 of the materials, stage 5 adds rand(1,3).
Brain trauma from dupping tc is gone.
Crystalized crossbread no longer say null upon inspecting.

Pylon no longer uses `process()` instead the `spawn_crystal()` is triggered on initialize and then every time one crystal is destroyed.

## Why It's Good For The Game

It was too good + fixes a bug

## Changelog
:cl:
tweak: Crystalized cerulean pylon only spawns 3 crystals at the time.
tweak: Each crystal takes twice as long to grow than before.
tweak: Each crystal only drops poly-crystal upon reaching state 4 (only 1 amt) or 5 (randomly chosen between 1 ~ 3 amt).
/:cl:

### CODE REVIEW

File: code/modules/research/xenobiology/crossbreeding/_structures.dm [hunk 1]
```diff
@@ -357,27 +361,46 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
...
-		crystal.amt = stage
+		if(stage == 5)
+			crystal.amt = rand(1,3)
+		else
+			crystal.amt = 1
+	if(pylon)
+		var/obj/structure/slime_crystal/cerulean/C = pylon.resolve()
+		if(C)
+			C.crystals--
+			C.spawn_crystal()
```
Comment (park66665) [inline code/modules/research/xenobiology/crossbreeding/_structures.dm:378]:
```suggestion
		if(C)
			C.crystals--
			C.spawn_crystal()
		else
			pylon = null
```

File: code/modules/research/xenobiology/crossbreeding/_structures.dm [hunk 2]
```diff
@@ -339,9 +339,13 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
 	max_integrity = 5
 	var/stage = 0
 	var/max_stage = 5
+	var/datum/weakref/pylon
 
 /obj/structure/cerulean_slime_crystal/Initialize()
 	. = ..()
+	var/obj/structure/slime_crystal/cerulean/master_pylon = locate(/obj/structure/slime_crystal/cerulean) in range(2, get_turf(src))
+	if(master_pylon)
+		pylon = WEAKREF(master_pylon)
```
Comment (park66665) [inline code/modules/research/xenobiology/crossbreeding/_structures.dm]:
```suggestion
/obj/structure/cerulean_slime_crystal/Initialize(mapload, obj/structure/slime_crystal/cerulean/master_pylon)
	. = ..()
	if(istype(master_pylon))
		pylon = WEAKREF(master_pylon)
```

File: code/modules/research/xenobiology/crossbreeding/_structures.dm [hunk 3]
```diff
@@ -357,27 +361,46 @@ GLOBAL_LIST_EMPTY(bluespace_slime_crystals)
...
+/obj/structure/slime_crystal/cerulean/proc/spawn_crystal()
+	if(crystals >= 3)
+		return
 	for(var/turf/T as() in RANGE_TURFS(2,src))
 		if(is_blocked_turf(T) || isspaceturf(T)  || T == get_turf(src) || prob(50))
 			continue
 		var/obj/structure/cerulean_slime_crystal/CSC = locate() in range(1,T)
 		if(CSC)
 			continue
 		new /obj/structure/cerulean_slime_crystal(T)
```
Comment (park66665) [inline code/modules/research/xenobiology/crossbreeding/_structures.dm]:
```suggestion
		new /obj/structure/cerulean_slime_crystal(T, src)
```

**Review by park66665 — COMMENTED** [review]
How about letting pylon pass a reference to itself when creating a new cerulean_slime_crystal instead of making cerulean_slime_crystal search its master pylon every time?

**Review by park66665 — APPROVED** [review]
As of now, this PR quite drastically cuts the amount of materials duplicated using crystallized cerulean. We'll see how this turns out.

All clear. This PR is good to go.

**Comment by BitBite112** [comment]
Archanial why did you do this. It's useless now. A maximum of 9 materials every 10 minutes it way too low. Pylons take a lot of space.

**Comment by RDS88-beeman** [comment]
Alright xenobiocord did the math and a crossbreed between two t1 slimes is now a straight upgrade in every way to the one between two t4 slimes. Game balance

**Comment by Archanial** [comment]
> 
> 
> Archanial why did you do this. It's useless now. A maximum of 9 materials every 10 minutes it way too low. Pylons take a lot of space.

This was consulted with our chief xenobio bounty maker (Ruko) and he said it was fine.

They take 4 tiles.

**Comment by BitBite112** [comment]
But they need to be spaced apart. It will block up any room significantly as well.

**Comment by BitBite112** [comment]
To match the productivity of what they used to be you're going to have to have 4.44 times as much pylons.

**Comment by BitBite112** [comment]
> To match the productivity of what they used to be you're going to have to have 4.44 times as much pylons.

Edit: I just noticed pylons used to be able to make more than 4 crystals, so you'll even need more pylons than the number I just gave to match their productivity

**Comment by Archanial** [comment]
That was the point of this PR. To limit its productivity.

**Comment by BitBite112** [comment]
But it's been nerfed to the ground now. It's not useful anymore time wise. I wouldn't mind making more if they didn't take so much space, there's just nowhere to put them without blocking all of the halls or walkingspaces.

**Pull request merged**